contracts/modules/metadata/CoreMetadataViewModule.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.23;
+
+import { IERC165 } from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
+import { Base64 } from "@openzeppelin/contracts/utils/Base64.sol";
+import { BaseModule } from "../BaseModule.sol";
+import { IIPAccount } from "../../interfaces/IIPAccount.sol";
+import { IPAccountStorageOps } from "../../lib/IPAccountStorageOps.sol";
+import { CORE_METADATA_VIEW_MODULE_KEY, CORE_METADATA_MODULE_KEY } from "../../lib/modules/Module.sol";
+import { ICoreMetadataViewModule, IViewModule } from "../../interfaces/modules/metadata/ICoreMetadataViewModule.sol";
+import { IModuleRegistry } from "contracts/interfaces/registries/IModuleRegistry.sol";
+
+/// @title Implementation of the ICoreMetadataViewModule interface
+/// @dev Provides functionalities to retrieve core metadata of IP assets, including name, description, and more.
+contract CoreMetadataViewModule is BaseModule, ICoreMetadataViewModule {
```

Comment:
To help frontends calling the contract directly, having a method that returns all the values, or inheriting Multicall could help

Reply:
Agreed! Added a function to retrieve all metadata.